### PR TITLE
[infra] dry cmake tests by introducing seqan3_test_component

### PR DESCRIPTION
**Pull request by marehr**
This PR will reduce some ugily and redundant cmake code.

follow up of #737

### CODE REVIEW

File: test/coverage/CMakeLists.txt [hunk 1]
```diff
@@ -76,32 +76,24 @@ add_custom_command(
...
-    #
-    # NOTE(marehr): ".+/test/" REGEX is greedy, that means
-    # /test/test/test/hello_test.cpp will result in an empty `test_path`
-    string (REGEX REPLACE "_test.cpp$" "" target_name ${target_cpp})
-    string (REGEX REPLACE ".+/test/" "" test_path ${CMAKE_CURRENT_LIST_DIR})
-    set (target "${target_name}_coverage_test")
-
-    add_executable (${target} ${target_cpp})
+macro (seqan3_test unit_test_cpp)
+    file (RELATIVE_PATH unit_test "${CMAKE_SOURCE_DIR}/.." "${CMAKE_CURRENT_LIST_DIR}/${unit_test_cpp}")
```
Comment (smehringer) [inline test/coverage/CMakeLists.txt]: Out of the three macros for performance, coverage and unit, only this one has the `/..` after souce dir ("${CMAKE_SOURCE_DIR}`/..`") why ?

File: test/seqan3-test.cmake [hunk 2]
```diff
@@ -91,6 +91,49 @@ list (APPEND SEQAN3_EXTERNAL_PROJECT_CMAKE_ARGS "-DCMAKE_INSTALL_PREFIX=${PROJEC
 # Commonly used macros for the different test modules in seqan3.
 # ----------------------------------------------------------------------------
 
+# e.g. target_source_file = "range/view/take.cpp"
+# component:
+#  * TARGET_NAME - the target name (e.g. "take")
+#  * TEST_NAME - the test name with target_path included (e.g. "range/view/take")
+#  * TARGET_PATH - the path to the target source (e.g. "range/view")
+macro (seqan3_test_component VAR _target_source_file _component_name_)
+    string (TOUPPER "${_component_name_}" _component_name)
```
Comment (smehringer) [inline test/seqan3-test.cmake]:
```suggestion
    string (TOUPPER "${component_name_}" component_name)
```
too many underscores to keep track of :eyes: 

You did not add underscores for e.g. `target_name` in the other cmake code at all so maybe we stick to it?

File: test/seqan3-test.cmake [hunk 3]
```diff
@@ -91,6 +91,49 @@ list (APPEND SEQAN3_EXTERNAL_PROJECT_CMAKE_ARGS "-DCMAKE_INSTALL_PREFIX=${PROJEC
 # Commonly used macros for the different test modules in seqan3.
 # ----------------------------------------------------------------------------
 
+# e.g. target_source_file = "range/view/take.cpp"
```
Comment (smehringer) [inline test/seqan3-test.cmake]: Add a sentence before the example of what the macro is doing.

**Comment by h-2** [comment]
Why are we refactoring the build system again when we have so many other things to do :sob:
It doesn't even reduce LOC...

File: test/coverage/CMakeLists.txt (hunk 1 above)
Comment (marehr) [reply test/coverage/CMakeLists.txt]: thank you! should have been `"${CMAKE_SOURCE_DIR}/../unit"`, because this is the base dir of the test cases.

**Pull request merged**